libs/git_wrapper/Git_wrapper.ml

```diff
@@ -631,6 +626,94 @@ let cat_file_blob ?cwd sha =
   | Error (`Msg s) ->
       Error s
 
+let obj_type_of_string = function
+  | "commit" -> Some Commit
+  | "blob" -> Some Blob
+  | "tree" -> Some Tree
+  | "tag" -> Some Tag
+  | _ -> None
+
+let batch_cat_file_blob ?cwd blob_shas =
+  let cmd =
+    Bos.Cmd.(
+      v "git"
+      %% of_list
+           (cd cwd
+           @ [
+               "cat-file";
+               (* Print object information and contents for each object
+                  provided on stdin. *)
+               "--batch";
+               (* Since we write to a file, ensure output is buffered. If we
+                  change to read from stdio, we probably should _not_ buffer so
+                  we can stream output. *)
+               "--buffer";
+               (* Null terminate the header line. Requires git >2.42.0. *)
+             ]))
+  in
+  (* Each object is formatted as
+
+         <oid> SP <type> SP <size> NUL
+         <contents> LF
+
+     See also
+     <https://git-scm.com/docs/git-cat-file#:~:text=For%20example%2C%20%2D%2Dbatch%20without%20a%20custom%20format%20would%20produce%3A>
+
+     This gets the next output out of the channel, and then leaves the rest.
+     This is so we can generate a lazy Seq.t so we don't have to read all of
```

Comment:
Well, I think the nature is documented via the type, unless I misunderstand you. I'll add an additional note about consuming it to prevent a resource leak.